### PR TITLE
Remove PHP backend.

### PR DESCRIPTION
**Pull request by treeform**
* I don't think PHP and JS are that similar, this makes JS gen confusing.
* PHP output was never advertised or really documented.

### CODE REVIEW

File: lib/system/gc_regions.nim [hunk 1]
```diff
@@ -8,23 +8,7 @@
 
 # "Stack GC" for embedded devices or ultra performance requirements.
 
-when defined(nimphpext):
```
Comment (Araq) [inline lib/system/gc_regions.nim]: This section is for writing PHP extensions with native Nim and should be kept.

File: compiler/jsgen.nim [hunk 2]
```diff
@@ -39,7 +38,7 @@ from modulegraphs import ModuleGraph
 
 type
   TTarget = enum
```
Comment (Yardanico) [inline compiler/jsgen.nim]: Is this really needed anymore? I'm not really into compiler code, but I think there's no point in this enum with only one possible value :)

Comment (treeform) [reply compiler/jsgen.nim]: you are right ill remove it.

File: lib/system/gc_regions.nim (hunk 1 above)
Comment (treeform) [reply lib/system/gc_regions.nim]: Ok keeping it.

**Pull request merged**